eth/core/state/statedb.go

```diff
@@ -150,15 +150,15 @@ func (sdb *stateDB) Finalise(bool) {
 // equivalent at the moment in Polaris as we do not leverage the state root.
 func (sdb *stateDB) IntermediateRoot(bool) common.Hash {
 	sdb.Finalise(true)
-	return common.Hash{}
+	return sdb.Plugin.StateRoot()
 }
 
 // Commit implements vm.PolarStateDB.
 func (sdb *stateDB) Commit(_ uint64, _ bool) (common.Hash, error) {
 	if err := sdb.Error(); err != nil {
 		return common.Hash{}, err
 	}
-	return common.Hash{}, nil
+	return sdb.Plugin.StateRoot(), nil
```

Comment:
this is returning the previous block header's hash, not the current one which was just committed